iroha/src/kura.rs

```diff
@@ -3,12 +3,23 @@ use async_std::{
     fs::{metadata, File},
     prelude::*,
 };
+use async_trait::async_trait;
 use std::{
     convert::TryFrom,
     fs,
     path::{Path, PathBuf},
 };
 
+#[async_trait]
+pub trait Storage {
+    fn new(_mode: String, block_store_path: &Path, world_state_view_tx: BlockSender) -> Self;
+
+    async fn init(&mut self) -> Result<(), String>;
+
+    /// Methods consumes new validated block and atomically stores and caches it.
+    async fn store(&mut self, mut block: Block) -> Result<Hash, String>;
+}
+
```

Comment:
Trait is not all public methods. Trait is a contract.
1. Please copy documentation to it from already written for `Kura`
2. Remove `new` and `init` methods because they are about implementation, not expectations.
3. Rewrite documentation for `store`: "Method stores a new valid block and propagate it to the World State View.\nReturns hash of the block stored if succeed or error message if failed.".